controllers/openshift/openshift.go

```diff
@@ -76,32 +76,14 @@ func reconcilerHook(cr *argoprojv1alpha1.ArgoCD, v interface{}, hint string) err
 				logv.Error(err, "failed to retrieve Cluster Role admin")
 				return err
 			}
-			policyRules := getPolicyRuleForApplicationController()
```

Comment:
This change gives the application controller additional permissions when run with the hooks being active.  Before the change, the verbs were restricted to `get, list, watch`, after the change, the verbs will be `*`.

Reply:
@jopit actually, the application-controller already has those permissions today
The application-controller policy rules are actually being defined in 2 different places 
They are first defined [here](https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/policyrule.go#L9) and assigned to the role during creation time. This already gives the application-controller permissions to do everything, and then when the reconcilerHook code is called, this 2nd set of permissions is appended to the existing rules but have no effect since they are a subset of what has already been provided to the role. 

I think whoever wrote this 2nd set of rules might have missed that these permissions had already been defined elsewhere

Reply:
according to this set of permissions, the application-controller would not be able to create any of the resources that are required for an application

Reply:
true

Reply:
An `admin` in OpenShift doesn't have * permissions for all the resources e.g deleting resource quotas. With these additional privileges, the application controller would be more than an `admin`

Reply:
> o.Rules = policyRules

@jaideepr97 I think the hook is overwriting these [permissions](https://github.com/argoproj-labs/argocd-operator/blob/master/controllers/argocd/policyrule.go#L9)

Reply:
Ah okay, I see what you're saying 
I wonder why those original permissions are even there then because it looks like they'll always get overwritten 
 how does the application-controller create any resources if it doesn't have `create` permissions then ?  :thinking:

Reply:
Ah okay, I see what you mean @chetan-rns 
maybe we should be deleting the other set of permissions then, since they'll always get overwritten by this more conservative set of permissions

Reply:
@jaideepr97 The hooks overwrite the permissions only on OpenShift environments. I think we still need to create the other set of permissions for Kubernetes.

Reply:
yeah, @chetan-rns I just followed the code and it looked like  `applyReconciler` was always getting called so I assumed the other set of permissions would always get overwritten
but it just occurred to me that we likely prevent this hook from getting registered at all when we package the argocd-operator